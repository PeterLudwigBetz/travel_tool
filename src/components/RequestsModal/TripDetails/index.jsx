import React, {PureComponent, Fragment} from 'react';
import {PropTypes} from 'prop-types';
import generateDynamicDate from '../../../helper/generateDynamicDate';
import TripDetail from './TripDetail';

export default class TripDetails extends PureComponent {
  static propTypes = {
    tripDetails: PropTypes.object.isRequired
  }
  renderTripDetails = (
    origin, destination, tripDetails, departureDate, tripType, returnDate, accomodationdetails
  ) => (
    <React.Fragment>
      <div>
        <TripDetail
          label="From"
          value={origin} />
      </div>
      <div>
        <TripDetail
          label="Travel to"
          value={destination} />
      </div>
      <div>
        <TripDetail
          label="Depature date"
          value={generateDynamicDate(tripDetails, departureDate)} />
      </div>
      <div id="return-date">
        { tripType !== 'oneWay' ?
          (<TripDetail
            label="Return date"
            value={generateDynamicDate(tripDetails, returnDate)} />
          )  : ''
        }
      </div>
      <div id="trip-detail">
        <TripDetail
          label="Accomodation"
          value={accomodationdetails} />
      </div>
    </React.Fragment>
  )
  renderTrips() {
    const { tripDetails } = this.props;
    const {
      beds,
      createdAt,
      departureDate,
      returnDate,
      destination,
      origin,
      tripType
    } = tripDetails;
    const {rooms:{roomName, guestHouses:{houseName}}, bedName } = beds;
    const accomodationdetails = `${houseName}, ${roomName}, ${bedName}`;
   
    return (
      <div className={`modal__modal-trip-details ${tripType}`}>
        {this.renderTripDetails(
          origin, destination, tripDetails, departureDate, tripType, returnDate, accomodationdetails
        )}
      </div>
    );
  }

  render() {
    return this.renderTrips();
  }
}
