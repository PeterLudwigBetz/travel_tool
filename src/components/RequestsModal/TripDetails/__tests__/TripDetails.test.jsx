import React from 'react';
import TripDetails from '..';

describe('<TripDetails />', () => {
  let wrapper, tripDetails;

  beforeEach(() => {
    tripDetails = {
      'origin': 'Lagos',
      'destination': 'Nairobi',
      'departureDate': '2018-12-09',
      'arrivalDate': '2018-12-11',
      'createdAt': '2018-08-15T11:11:52.181Z'
    };
    wrapper = mount(
      <TripDetails tripDetails={tripDetails} />
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
