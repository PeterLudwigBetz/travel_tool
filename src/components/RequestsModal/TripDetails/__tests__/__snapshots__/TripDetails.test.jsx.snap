// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripDetails /> should render correctly 1`] = `
<div
  className="modal__modal-trip-details undefined"
>
  <div
    className="modal__trip-detail"
  >
    <div
      className="modal__trip-detail-label"
    >
      From
    </div>
    <div
      className="modal__trip-detail-text"
    >
      Lagos
    </div>
  </div>
  <div
    className="modal__trip-detail"
  >
    <div
      className="modal__trip-detail-label"
    >
      Travel to
    </div>
    <div
      className="modal__trip-detail-text"
    >
      Nairobi
    </div>
  </div>
  <div
    className="modal__trip-detail"
  >
    <div
      className="modal__trip-detail-label"
    >
      Depature date
    </div>
    <div
      className="modal__trip-detail-text"
    >
      09 Dec 2018
    </div>
  </div>
  <div
    id="return-date"
  >
    <div
      className="modal__trip-detail"
    >
      <div
        className="modal__trip-detail-label"
      >
        Return date
      </div>
      <div
        className="modal__trip-detail-text"
      >
        04 Oct 2018
      </div>
    </div>
  </div>
</div>
`;
